### PR TITLE
Add java doc based on sentry-data-schemes project

### PR DESCRIPTION
**Pull request by Panthaaaa**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
Added java docs to the protocol classes based on the https://github.com/getsentry/sentry-data-schemas project.

## :bulb: Motivation and Context
It improves accessibility of the SDK because all the important information is directly in the javadoc instead of in someones browser. 

## :green_heart: How did you test it?
`make compile` as I only added java docs.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/protocol/User.java [hunk 1]
```diff
@@ -9,22 +9,30 @@
...
-  /** User's id */
+  /** Unique identifier of the user. */
   private @Nullable String id;
 
-  /** User's username */
+  /** Username of the user. */
   private @Nullable String username;
 
-  /** User's ipAddress */
+  /** Remote IP address of the user. Defaults to "{{auto}}". */
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/protocol/User.java:29]:
```suggestion
  /** Remote IP address of the user. */
```

Comment (marandaneto) [reply sentry/src/main/java/io/sentry/protocol/User.java:29]: defaults after #1015

File: sentry/src/main/java/io/sentry/protocol/Request.java [hunk 2]
```diff
@@ -4,14 +4,92 @@
...
+   *
+   * <p>If the query string is not declared and part of the `url`, Sentry moves it to the query
+   * string.
+   */
   private String queryString;
+  /**
+   * Request data in any format that makes sense.
+   *
+   * <p>SDKs should discard large and binary bodies by default. Can be given as string or structural
+   * data of any format.
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/protocol/Request.java:67]:
```suggestion
   * data of any serializable format.
```

Comment (marandaneto) [reply sentry/src/main/java/io/sentry/protocol/Request.java:67]: just to be clear that it needs to be serializable

File: sentry/src/main/java/io/sentry/SentryEvent.java [hunk 3]
```diff
@@ -1,44 +1,160 @@
 package io.sentry;
 
 import io.sentry.protocol.*;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/SentryEvent.java:4]: I'd keep the specific imports

**Review by marandaneto — APPROVED** [review]
thanks for doing this @Panthaaaa appreciated.

I've written 2 or 3 comments but they are easy fixes, work on them and LGTM.

**Comment by marandaneto** [comment]
`./gradlew spotlessApply` would fix the broken CI :)

**Review by maciejwalkowiak — APPROVED** [review]
This is awesome! Thanks a lot for this PR.

**Comment by Panthaaaa** [comment]
> `./gradlew spotlessApply` would fix the broken CI :)

It fixed it, but its conflicting with the editorconfig. (So IDEA reverted it on commit, but I think thats why you recommend the commit-hook)

**Pull request merged**